linalg/complex_operator.hpp

```diff
@@ -168,6 +174,69 @@ class ComplexSparseMatrix : public ComplexOperator
    virtual Type GetType() const { return MFEM_ComplexSparseMat; }
 };
 
+#ifdef MFEM_USE_SUITESPARSE
+/** @brief Interface with UMFPack solver specialized for ComplexSparseMatrix
+    This approach avoids forming a monolithic SparseMatrix which leads
+    to increased memory and flops
+ */
+class ComplexUMFPackSolver : public Solver
+{
+protected:
+   bool use_long_ints;
+   bool transa = false;
+   ComplexSparseMatrix *mat;
+
+   void *Numeric;
+   SuiteSparse_long *AI, *AJ;
+
+   void Init();
+
+public:
+   double Control[UMFPACK_CONTROL];
+   mutable double Info[UMFPACK_INFO];
+
+   /** @brief For larger matrices, if the solver fails, set the parameter @a
+       _use_long_ints = true. */
+   ComplexUMFPackSolver(bool _use_long_ints = false)
+      : use_long_ints(_use_long_ints) { Init(); }
+   /** @brief Factorize the given ComplexSparseMatrix using the defaults.
+       For larger  matrices, if the solver fails, set the parameter
+       @a _use_long_ints = true. */
+   ComplexUMFPackSolver(ComplexSparseMatrix &A, bool _use_long_ints = false)
+      : use_long_ints(_use_long_ints) { Init(); SetOperator(A); }
+
+   /** @brief Factorize the given Operator @a op which must be
+       a ComplexSparseMatrix.
+
+       The factorization uses the parameters set in the #Control data member.
+       @note This method calls SparseMatrix::SortColumnIndices()
+       for real imag parts of the ComplexSparseMatrix,
+       modifying the matrices if the column indices are not already sorted. */
+   virtual void SetOperator(const Operator &op);
+
+   // Set the print level field in the #Control data member.
+   void SetPrintLevel(int print_lvl) { Control[UMFPACK_PRL] = print_lvl; }
+
+   void SetTransposeSolve(bool transa_) { transa = transa_ ;}
```

Comment:
- Can you add a comment here so it's clear that setting this member variable only affects the call to `MultTranspose` (referring to the explanation there), but has no effect on `Mult`.
- Space missing after semi-colon.

Reply:
Done